### PR TITLE
add test for encode on union property

### PR DESCRIPTION
**Pull request by chunyu3**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/encode/union/main.tsp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for encode decorator on union property.")
+@supportedBy("dpg")
+@scenarioService("/encode/union")
+namespace Encode.Union;
+
+model UnixTimestampDatetimeUnionProperty {
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/encode/union/main.tsp:13]: I don't think this be a union property (as I don't think this is still valid if type be `utcDateTime | string`). Seems better to call it nullable property.

UnixTimestampDatetimeNullableProperty ?

**Review by weidongxu-microsoft — APPROVED** [review]

File: packages/cadl-ranch-specs/http/encode/union/main.tsp (hunk 1 above)
Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/encode/union/main.tsp:13]: Should it just be moved into https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/encode/datetime/main.tsp ?

Comment (tadelesh) [reply packages/cadl-ranch-specs/http/encode/union/main.tsp:13]: agree

**Comment by timotheeguerin** [comment]
doing some cleanup as we are migrating scenarios to typespec and typespec-azure repo, please repopen if it is still relevant. If it is please make equivalent change in microsoft/typespec

**Pull request closed without merging**